RELEASE_PROCEDURE.md

```diff
@@ -0,0 +1,36 @@
+# Release procedure for DLA-Future
+
+DLA-Future follows [Semantic Versioning](https://semver.org).
+
+1. For minor and major releases: create and check out a new branch at an
+   appropriate point on `main` with the name `release-major.minor`.  `major` and `minor` should be the
+   major and minor versions of the release. For patch releases: check out the corresponding
+   `release-major.minor` branch.
+
+1. Write release notes in `CHANGELOG.md`. Check for issues and pull requests for the release on the
+   [DLA-F Planning board](https://github.com/orgs/eth-cscs/projects/1). All list entries and paragraphs
+   must be on a single line for correct formatting on GitHub.
```

Comment:
I would add a note like "Only relevant changes should be mentioned in the changelog."

Reply:
Do you have a definition for "relevant changes"? Only user-facing changes? Only major changes? Something else?

Reply:
A sort of guideline (but for sure non exhaustive) can be

I excluded:
 - anything that was related to CI and github actions
 - anything related to spack package (Note: change of dependencies are not stated as spack package change)
 - documentation related changes
 - anything related to plotting scripts
Must be included:
 - changes in user API
 - new functionalities
 - bugfixes
 - performance critical changes
 - change in dependencies

Reply:
How does this look: https://github.com/eth-cscs/DLA-Future/pull/1127/commits/070aa621075f9d0d305294304f0297e407b7ef66?